k9mail/src/main/java/com/fsck/k9/preferences/SettingsImporter.java

```diff
@@ -935,11 +942,8 @@ private static ImportedAccount parseAccount(XmlPullParser xpp, List<String> acco
                             account.settings = parseSettings(xpp, SettingsExporter.SETTINGS_ELEMENT);
                         }
                     } else if (SettingsExporter.IDENTITIES_ELEMENT.equals(element)) {
-                        if (overview) {
-                            skipToEndTag(xpp, SettingsExporter.IDENTITIES_ELEMENT);
-                        } else {
-                            account.identities = parseIdentities(xpp);
-                        }
+                        //Read the full content to have an alternative identifier to display if the name is empty
```

Comment:
We try to avoid comments. This particular information is nice to have in a commit message, though.

The functional aspect is best checked in a test. So if someone were to change the code back to its previous form the test would tell us something is wrong rather than a comment that can be easily overlooked, ignored, or is out of date.